packages/chat-client/src/controllers/engine.ts

```diff
@@ -496,25 +498,29 @@ export class ChatEngine extends IChatEngine {
       ),
     };
 
-    await this.sendRequest(messagePayload.topic, "wc_chatMessage", {
+    const jsonRpcPayload = formatJsonRpcRequest("wc_chatMessage", {
       messageAuth: await this.generateIdAuth(
         this.currentAccount,
         messageKeyClaims
       ),
     });
 
-    console.log("----- SEND MSG");
+    const encodedMessage = await this.client.core.crypto.encode(
+      messagePayload.topic,
+      jsonRpcPayload
+    );
 
-    // const {
-    //   done: acknowledged,
-    //   resolve,
-    //   reject,
-    // } = createDelayedPromise<void>();
-    // this.events.once(engineEvent("chat_message", id), ({ error }) => {
-    //   if (error) reject(error);
-    //   else resolve();
-    // });
-    // await acknowledged();
+    await this.client.core.relayer.provider.request({
+      method: getRelayProtocolApi(getRelayProtocolName().protocol).publish,
+      params: {
+        ...ENGINE_RPC_OPTS["wc_chatMessage"].req,
+        message: encodedMessage,
+        topic: messagePayload.topic,
+      },
+    });
+    this.client.core.history.set(messagePayload.topic, jsonRpcPayload);
```

Comment:
Ordering here is different compared to `sendRequest`, which sets the `history` record first, but I'm guessing this is intentional to avoid setting a record before the request promise has resolved, i.e. `this.setMessage` is also called afterwards.

Reply:
Precisely